src/docs/frontend/using-styled-components.mdx

```diff
@@ -0,0 +1,221 @@
+
+const Numeric = styled('span')`
+  font-variant-numeric: tabular-nums;
+`;
+
+const SideBySide = styled('div')`
+  display: flex;
+  gap: ${space(2)};
+  flex-wrap: wrap;
+  align-items: flex-start;
+`;
+
+export default ExampleComponent() {
+  return (
+    <SideBySide>
+      <p>a number: <Numeric>1</Numeric></p>
+      <p>and another: <Numeric>2</Numeric></p>
+    </SideBySide>
+  )
+}
+```
+
+## Theme
+
+Take constants (z-indexes, paddings, colors) from [props.theme](https://github.com/getsentry/sentry/blob/master/static/app/utils/theme.tsx)
+
+```tsx
+import styled from '@emotion/styled';
+import {space} from 'sentry/styles/space';
+
+const SomeComponent = styled('div')`
+  border-radius: 1.45em;
+  font-weight: bold;
+  z-index: ${p => p.theme.zIndex.modal};
+  padding: ${space(1)} ${space(2)};
+  border: 1px solid ${p => p.theme.borderLight};
+  color: ${p => p.theme.purple};
+  box-shadow: ${p => p.theme.dropShadowHeavy};
+`;
+```
+
+## Dynamic Properties
+
+<Alert level="info">
+  Most of the time dynamic properties are not needed!
+</Alert>
+
+You can pass in custom props to your custom component and use that to set styles. This does create overhead at runtime and should be avoided. What follows is an example of how to pass custom props, and how to avoid doing so.
+
+### Avoid Dynamic Props
+```tsx
+import styled from '@emotion/styled';
+
+// ❌ Try to avoid this
+const Label = styled('label')<{isDisabled: boolean; isError: boolean}>`
+  color: ${p => p.isDisabled
+    ? p.theme.gray200
+    : p.isError
+    ? p.theme.error400
+    : inherit
+  };
+`;
+
+return <Label disabled={false} isError={true}>There is an error</Label>;
+```
+
+Choose one of the strategies below instead.
+
+### Use Data Selectors
+
+Data selectors can be an easy way to set styles based on simple/pre-defined state. When defining these attributes don't forget that browser have built-in selectors like `:disabled`, `:focus`, `:first-child` and so on.
+
+```tsx
+import styled from '@emotion/styled';
+
+// ✅ Use a data attribute to pass in your state, and leverage CSS selectors:
+const Label = styled('label')`
+  color: inherit;
+  &[data-is-error="true"] {
+    color: ${p => p.theme.error400};
+  }
+  
+  &:disabled {
+    color: ${p => p.theme.gray200};
+  }
+`;
+
+return <Label disabled={false} data-is-error={true}>There is an error</Label>;
+```
+
+### Use CSS Variables
+
+For more flexible values, you can use CSS variables instead of pre-defined data-attributes for some value types.
+
+```tsx
+import styled from '@emotion/styled';
+
+// ✅ Define css variables for overridable properties:
+const Label = styled('label')`
+  color: var(--color, ${p => p.theme.textColor});
+`;
+
+const Error = styled('div')`
+  --color: ${p => p.theme.red400};
+`;
+const Info = styled('div')`
+  --color: ${p => p.theme.blue400};
+`;
+
+const label = <Label>There is an error</Label>;
+return isError
+  ? <Error>{label}</Error>
+  : <Info>{label}</Info>;
```

Comment:
I like this approach but the docs confused me a bit. This would work without CSS variables through the cascade, right? This example feels like it's more about using conditional rendering (choosing the correct wrapper) than about CSS variables

Reply:
yeah, i'll take this section out. it's not a strong example of what i was going for... maybe better in a future 'how to override/subclass components' section.